src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java

```diff
@@ -69,8 +80,38 @@ public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) throws I
 
             @Override
             public void onResponse(MLCreateConnectorResponse mlCreateConnectorResponse) {
-                logger.info("Created connector successfully");
-                // TODO Add the response to Global Context
+                try {
+                    logger.info("Created connector successfully");
+                    String workflowId = data.get(0).getWorkflowId();
+                    String workflowStepName = getName();
+                    ResourcesCreated newResource = new ResourcesCreated(workflowStepName, mlCreateConnectorResponse.getConnectorId());
+                    XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                    newResource.toXContent(builder, ToXContentObject.EMPTY_PARAMS);
+
+                    // The script to append a new object to the resources_created array
+                    Script script = new Script(
+                        ScriptType.INLINE,
+                        "painless",
+                        "ctx._source.resources_created.add(params.newResource)",
+                        Collections.singletonMap("newResource", newResource)
+                    );
```

Comment:
@joshpalis can you verify the painless script here?

Reply:
Will have more tests when adding created resources is fully complete